### PR TITLE
chore: update Observability Operator to v3.0.10

### PR DESCRIPTION
**Pull request by vmanley**
## Description

Jira: [MGDSTRM-8567](https://issues.redhat.com/browse/MGDSTRM-8567)
This change updates the version of the Observability Operator installed by KAS Fleet Manager from v3.0.9 to v3.0.10
## Verification Steps

1. Run KAS Fleet Manager locally.
2. The Observability Operator should be deployed in the `managed-application-services-observability` namespace.
3. Check `observability-operator-controller-manager` YAML for `spec/containers/image`. This value should be `quay.io/rhoas/observability-operator:v3.0.10`

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [ ] ~~Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)~~
- [ ] ~~Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] ~~Required metrics/dashboards/alerts have been added (or PR created).~~
- [ ] ~~Required Standard Operating Procedure (SOP) is added.~~
- [ ] ~~JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: internal/kafka/internal/workers/clusters_mgr.go [hunk 1]
```diff
@@ -991,7 +991,7 @@ func (c *ClusterManager) buildObservabilitySubscriptionResource() *v1alpha1.Subs
 			CatalogSource:          observabilityCatalogSourceName,
 			Channel:                "alpha",
 			CatalogSourceNamespace: observabilityNamespace,
-			StartingCSV:            "observability-operator.v3.0.9",
+			StartingCSV:            "observability-operator.v3.0.10",
```
Comment (machi1990) [inline internal/kafka/internal/workers/clusters_mgr.go:994]: maybe its time we make this configurable via cli flags

File: internal/kafka/internal/workers/clusters_mgr.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ import (
 
 const (
 	observabilityNamespace          = "managed-application-services-observability"
-	observabilityCatalogSourceImage = "quay.io/rhoas/observability-operator-index:v3.0.9"
+	observabilityCatalogSourceImage = "quay.io/rhoas/observability-operator-index:v3.0.10"
```
Comment (machi1990) [inline internal/kafka/internal/workers/clusters_mgr.go:41]: maybe its time we make this configurable via cli flags

**Review by machi1990 — CHANGES_REQUESTED** [review]

File: internal/kafka/internal/workers/clusters_mgr.go (hunk 2 above)
Comment (vmanley) [reply internal/kafka/internal/workers/clusters_mgr.go:41]: I can make a Jira to make this change. Thanks @machi1990

File: internal/kafka/internal/workers/clusters_mgr.go (hunk 1 above)
Comment (vmanley) [reply internal/kafka/internal/workers/clusters_mgr.go:994]: I can make a Jira to make this change. Thanks @machi1990

File: internal/kafka/internal/workers/clusters_mgr.go (hunk 2 above)
Comment (machi1990) [reply internal/kafka/internal/workers/clusters_mgr.go:41]: That will be perfect, thanks @vmanley

Comment (vmanley) [reply internal/kafka/internal/workers/clusters_mgr.go:41]: Jira to make OO version configurable: https://issues.redhat.com/browse/MGDSTRM-8588

Comment (machi1990) [reply internal/kafka/internal/workers/clusters_mgr.go:41]: Thanks

**Review by machi1990 — APPROVED** [review]

**Pull request merged**